cmd/codegen/util/files.go

```diff
@@ -12,6 +12,58 @@ import (
 	"github.com/pkg/errors"
 )
 
+// BackupGoModAndGoSum creates backup copies of go.mod and go.sum and returns a function to run at the end of execution
+// to revert the go.mod and go.sum in the repo to the backups.
+func BackupGoModAndGoSum() (func(), error) {
```

Comment:
that's to avoid that the actual code generation makes changes to the go modules, right? But is the actual code generation in this case still "versioned"?

Reply:
Yup.

Reply:
This is so that we can use go modules for the generation part (opposed to turning off go-modules). ATM, it would update the go.mod file which we don't want, since versions should only be driven by the runtime dependencies.

Reply:
I am kind of undecided whether this is a good idea or not, but I don't really see an alternative.

Reply:
Same here.